library/core/src/mem/mod.rs

```diff
@@ -304,7 +304,6 @@ pub fn forget_unsized<T: ?Sized>(t: T) {
 #[inline(always)]
 #[must_use]
 #[stable(feature = "rust1", since = "1.0.0")]
-#[rustc_promotable]
 #[rustc_const_stable(feature = "const_mem_size_of", since = "1.24.0")]
 #[cfg_attr(not(test), rustc_diagnostic_item = "mem_size_of")]
 pub const fn size_of<T>() -> usize {
```

Comment:
`size_of` and `align_of` are effectively constants, so it seems fine to keep allowing them

```
error[E0515]: cannot return value referencing temporary value
[INFO] [stdout]   --> /opt/rustwide/cargo-home/registry/src/github.com-1ecc6299db9ec823/pixel_engine_backend-0.4.1/src/lib.rs:27:9
[INFO] [stdout]    |
[INFO] [stdout] 27 |  /         wgpu::VertexBufferLayout {
[INFO] [stdout] 28 |  |             array_stride: mem::size_of::<Vertex>() as wgpu::BufferAddress,
[INFO] [stdout] 29 |  |             step_mode: wgpu::VertexStepMode::Vertex,
[INFO] [stdout] 30 |  |             attributes: &[
[INFO] [stdout]    |  |__________________________-
[INFO] [stdout] 31 | ||                 wgpu::VertexAttribute {
[INFO] [stdout] 32 | ||                     offset: 0,
[INFO] [stdout] 33 | ||                     shader_location: 0,
[INFO] [stdout] ...  ||
[INFO] [stdout] 45 | ||                 },
[INFO] [stdout] 46 | ||             ],
[INFO] [stdout]    | ||_____________- temporary value created here
[INFO] [stdout] 47 |  |         }
[INFO] [stdout]    |  |_________^ returns a value referencing data owned by the current function
```